crypto/keys/secp256k1/secp256k1.go

```diff
@@ -21,16 +21,74 @@ import (
 
 var (
 	_ cryptotypes.PrivKey  = &PrivKey{}
+	_ cryptotypes.PrivKey  = &PrivKeyOld{}
 	_ codec.AminoMarshaler = &PrivKey{}
+	_ codec.AminoMarshaler = &PrivKeyOld{}
 )
 
 const (
-	PrivKeySize = 32
-	keyType     = "secp256k1"
-	PrivKeyName = "tendermint/PrivKeySecp256k1"
-	PubKeyName  = "tendermint/PubKeySecp256k1"
+	PrivKeySize    = 32
+	keyType        = "secp256k1"
+	PrivKeyNameOld = "tendermint/PrivKeySecp256k1"
+	PubKeyNameOld  = "tendermint/PubKeySecp256k1"
+	PrivKeyName    = "comet/PrivKeySecp256k1Uncompressed"
+	PubKeyName     = "comet/PubKeySecp256k1Uncompressed"
 )
 
+func (privKey *PrivKeyOld) Bytes() []byte {
+	return privKey.Key
+}
+func (privKey *PrivKeyOld) PubKey() cryptotypes.PubKey {
+	privateObject, err := ethCrypto.ToECDSA(privKey.Key)
+	if err != nil {
+		panic(err)
+	}
+
+	pk := ethCrypto.FromECDSAPub(&privateObject.PublicKey)
+	return &PubKey{Key: pk}
+}
+func (privKey *PrivKeyOld) Equals(other cryptotypes.LedgerPrivKey) bool {
+	return privKey.Type() == other.Type() && subtle.ConstantTimeCompare(privKey.Bytes(), other.Bytes()) == 1
+}
+func (privKey *PrivKeyOld) Type() string {
+	return keyType
+}
+func (privKey *PrivKeyOld) Sign(msg []byte) ([]byte, error) {
```

Comment:
`PrivKey` doesn't have method `Sign`. Do we need it here?

Reply:
It is there, but not in the same file. I kept as it was originally for `comos-sdk`, meaning `crypto/keys/secp256k1/secp256k1_nocgo.go`, see [here](https://github.com/0xPolygon/cosmos-sdk/pull/8/files#diff-30a6b594e157a12a28bf8a26f65ab4de1dc3c65c8419e5756f0b9c25d9ce1a93R17)